packages/picker/src/Picker.ts

```diff
@@ -253,27 +253,34 @@ export class PickerBase extends SizedMixin(Focusable) {
 
     private async openMenu(): Promise<void> {
         /* c8 ignore next 9 */
-        let reparentableChildren: Element[] = [];
+        let reparentableChildren: MenuItem[] = [];
 
         const deprecatedMenu = this.querySelector('sp-menu');
         if (deprecatedMenu) {
-            reparentableChildren = Array.from(deprecatedMenu.children);
+            reparentableChildren = Array.from(
+                deprecatedMenu.children
+            ) as MenuItem[];
         } else {
             reparentableChildren = Array.from(this.children).filter(
                 (element) => {
                     return !element.hasAttribute('slot');
                 }
-            );
+            ) as MenuItem[];
```

Comment:
Won't this break re-parenting of MenuDividers? There's a separate query selector to get the menu items (which also might be problematic for sub-classing, I see).

Assuming this does break for MenuDivider though, we should add a test!

Reply:
(it does seem to work on your branch, but this code still looks wrong)

Reply:
I think it looks wrong, because it was needed for something else, but I've backed out of it 🙈

I should be able to revert this to `Element[]`.

Reply:
this is why code reviews are good! It definitely felt like there was more to the story.